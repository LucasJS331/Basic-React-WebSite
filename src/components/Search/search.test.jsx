import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { Search } from "."

describe("<Search />", ()=>{
    it("shoud be in document", ()=>{
        let fn = jest.fn();
        render(<Search placeholder={"search here"} onChange={fn} value={"test"} />);
        expect.assertions(1);
        const search = screen.getByPlaceholderText("search here");
        expect(search).toBeInTheDocument();

    })

    it("shoud be call function on key press", ()=>{
        let fn = jest.fn();
        render(<Search placeholder={"search here"} onChange={fn} value={"test"} />);

        let search = screen.getByPlaceholderText("search here");
        userEvent.type(search, "dev");

        expect(fn).toBeCalledTimes(3);
    })

    it("should be match snapshot", ()=>{
        let fn = jest.fn();
        let {container} = render(<Search placeholder={"search here"} onChange={fn} value={"test"} />);
        expect(container.firstChild).toMatchSnapshot();


    })
})
