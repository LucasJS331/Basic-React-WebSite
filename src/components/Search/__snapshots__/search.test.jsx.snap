// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should be match snapshot 1`] = `
<input
  class="input"
  placeholder="search here"
  type="search"
  value="test"
/>
`;
